### PR TITLE
Add the primitive print

### PR DESCRIPTION
**Pull request by DarkDrek**
Implements #44 

The new implementation is a lot cleaner the only bad place is inside `type_of` without `box_patterns` it isn't easy to match on the Box content. I did it the simplest way and do not validate the args of the call but that may be necessary for other primitives. They could for example have the ability to change their return type depending on the argument type.

PS: I found an error in the Display impl for the error type and fixed it in this PR but can split it if necessary.

### CODE REVIEW

File: src/ty/ty_check.rs [hunk 1]
```diff
@@ -156,7 +161,7 @@ impl<'a> Context<'a> {
                 }
             }
             Term::PrimFn(prim) => match prim {
-                _ => todo!("There aren't any primitives yet!"),
+                Primitive::Print => Ty::Unit,
```
Comment (pvdrz) [inline src/ty/ty_check.rs]: Maybe we can fix this in the same way as @seanchen1991 did before we had binary operations as `Term`s inside the MIR.

You can add an special case to the rule for `Term::App(t1, t2)` when `t1` is `Term::PrimFn(Primitive::Print)` and type it as `Ty::Unit`. 

And then you can change this arm to `unreachable!()`

Comment (DarkDrek) [reply src/ty/ty_check.rs]: Moved type check for primitives to own method

**Review by pvdrz — APPROVED** [review]

File: src/parser/node/mod.rs [hunk 2]
```diff
@@ -72,6 +73,7 @@ fn base_node(input: Span) -> IResult<Located<Node>> {
         }),
         lookahead(pair(tag("if"), multispace1), cond::cond),
         lookahead(tag("fn"), fn_def::fn_def),
+        lookahead(primitive, call::call),
```
Comment (pvdrz) [inline src/parser/node/mod.rs:76]: you made me wonder. What happens with the parser if you try to parse `(print)(10)`?

**Review by pvdrz — APPROVED** [review]
I'm pretty happy with this :)

**Pull request merged**